packages/dev/core/src/Materials/Textures/Procedurals/proceduralTexture.ts

```diff
@@ -694,18 +694,30 @@ export class ProceduralTexture extends Texture {
                 engine.drawElementsType(Material.TriangleFillMode, 0, 6);
             }
         } else {
-            engine.bindFramebuffer(this._rtWrapper, 0, undefined, undefined, true);
+            let numLayers = 1;

```

Comment:
If the RT is 3D or a 2D Array, we loop through the number of layers and render each.
For a regular, 2D texture, we render layer 0 and don't set the layer uniform.